### PR TITLE
Add `storage_graph` to show the plugins stored or needed to be calculated in the dependency tree

### PR DESCRIPTION
**Pull request by dachengx**
_Before you submit this PR: make sure to put all operations-related information in a wiki-note, a PR should be about code and is publicly accessible_

## What does the code in this PR do / what does it improve?

Depends on https://github.com/XENONnT/base_environment/pull/1593

[slack thread](https://xenonnt.slack.com/archives/C016UJZ090B/p1712323462029059)

As an example:
```
import straxen
st = straxen.contexts.demo()
test_run_id_1T = "180423_1021"
st.storage_graph(test_run_id_1T, "event_info", format="png")
```
gives you file `180423_1021-event_info.gv.png`:

event_info gv

## Can you briefly describe how it works?

1. green means already stored
2. orange means `strax.SaveWhen.ALWAYS` so that it will be saved if you process raw_records
3. yellow means `strax.SaveWhen.TARGET`
4. red means `strax.SaveWhen.EXPLICIT`
5. grey means `strax.SaveWhen.NEVER`

## Can you give a minimal working example (or illustrate with a figure)?

_Please include the following if applicable:_
  - [ ] _Update the docstring(s)_
  - [ ] _Update the documentation_
  - [ ] _Tests to check the (new) code is working as desired._
  - [ ] _Does it solve one of the open issues on github?_

### _Notes on testing_
 - _Until the automated tests pass, please mark the PR as a draft._
 - _On the XENONnT fork we test with database access, on private forks there is no database access for security considerations._

All _italic_ comments can be removed from this template.

### CODE REVIEW

**Review by yuema137 — DISMISSED** [review]
@dachengx Thanks for this PR! I tested on several plugins and they look good. I just added the color list to docstring as it may not be explicit for the users.

File: straxen/misc.py [hunk 1]
```diff
@@ -393,3 +396,87 @@ def sizeof(o):
...
+    :param to_dir: str, directory to save the plot
+    :param format: str, format of the plot
+    :return: all plugins that will be calculated when running self.make(run_id, target)
+
+    The colors used in the graph represent the following storage states:
+    - Grey: strax.SaveWhen.NEVER
+    - Red: strax.SaveWhen.EXPLICIT
+    - Orange: strax.SaveWhen.TARGET
+    - Yellow: strax.SaveWhen.ALWAYS
+    - Green: Plugin is stored
```
Comment (dachengx) [inline straxen/misc.py:427]: I think here `target is stored`

Comment (yuema137) [reply straxen/misc.py:427]: ok let me update

**Review by yuema137 — APPROVED** [review]

**Comment by dachengx** [comment]
The failure of test is a comer case that might be fixed in another PR. https://github.com/XENONnT/straxen/actions/runs/8576757387/job/23508277771?pr=1353

**Pull request merged**